src/prefect/workers/base.py

```diff
@@ -1562,6 +1591,267 @@ async def _mark_flow_run_as_cancelled(
                 f"Flow run '{flow_run.id}' was deleted before it could be marked as cancelled"
             )
 
+    def _is_cancellation_enabled(self) -> bool:
+        """Check if worker-side cancellation handling is enabled."""
+        return get_current_settings().worker.enable_cancellation
+
+    async def _setup_cancellation_handling(self) -> None:
+        """Initialize cancellation detection via WebSocket subscription."""
+        try:
+            subscriber = get_events_subscriber(
+                filter=EventFilter(
+                    event=EventNameFilter(name=["prefect.flow-run.Cancelling"])
+                ),
+            )
+            self._cancellation_events_subscriber = (
+                await self._exit_stack.enter_async_context(subscriber)
+            )
+        except Exception as exc:
+            self._logger.warning(
+                "Failed to set up cancellation event subscriber. "
+                "Pending flow runs will not be cancelled automatically. Error: %s",
+                exc,
+            )
+            return
+
+        # Poll once on startup for any orphaned cancelling flow runs
+        try:
+            await self._poll_for_cancelling_flow_runs()
+        except Exception as exc:
+            self._logger.warning(
+                "Failed to poll for cancelling flow runs on startup: %s", exc
+            )
+
+        # Start the WebSocket consumer task
+        self._cancellation_consumer_task = asyncio.create_task(
+            self._consume_cancellation_events(),
+            name="cancellation-event-consumer",
+        )
+        self._cancellation_consumer_task.add_done_callback(
+            self._handle_cancellation_consumer_done
+        )
+
+    def _handle_cancellation_consumer_done(self, task: asyncio.Task[None]) -> None:
```

Comment:
The `BaseWorker` class already has a ton of methods, so I think it'd be worthwhile to create a new class to encapsulate the new cancellation logic. Something like the `FlowRunCancellingObserver` that the `Runner` class uses would work well here.

Reply:
Yeah, that makes sense. I wonder if we have a stand alone class for cancellation, I would think we would want to re-use that in the runner as well... That is likely out of scope for this initial change.

Reply:
I was able to use the `FlowRunCancellingObserver` directly with a minor tweak to the EventFilter